api/services/tools/builtin_tools_manage_service.py

```diff
@@ -16,13 +16,24 @@
 from core.tools.tool_manager import ToolManager
 from core.tools.utils.configuration import ProviderConfigEncrypter
 from extensions.ext_database import db
+from extensions.ext_redis import redis_client
 from models.tools import BuiltinToolProvider
 from services.tools.tools_transform_service import ToolTransformService
 
 logger = logging.getLogger(__name__)
 
 
 class BuiltinToolManageService:
+    REDIS_KEY_PREFIX = "builtin_tools:user:{user_id}:tenant:{tenant_id}"
+    REDIS_TTL = 60 * 10  # Cache expires in 10 minutes
+
+    @staticmethod
+    def get_redis_key(user_id: str, tenant_id: str) -> str:
```

Comment:
```suggestion
    def generate_cache_key(user_id: str, tenant_id: str) -> str:
```
prefer the netual concept `cache` instead of specific implementation `redis`.